preprocessors/op-street-map.schema.json

```diff
@@ -0,0 +1,158 @@
+                                "properties": {
+                                    "street_id": {"type": "number"},
+                                    "street_name": {"type": "string"},
+                                    "street_type": {"type": "string"},
+                                    "addr:street": {"type": "string"},
+                                    "surface": {"type": "string"},
+                                    "oneway": {"type": "boolean"},
+                                    "sidewalk": {"type": "string"},
+                                    "maxspeed": {"type": "string"},
+                                    "lanes": {"type": "number"},
+                                    "nodes": {
+                                        "type": "array",
+                                        "title": "nodes",
+                                        "description": " An ordered list of nodes along the respective street (way)",
+                                        "items": {
+                                            "type": "object",
+                                            "title": "node",
+                                            "description": "A single point in space defined by its latitude, longitude and node ID in the OpenStreetMap (https://wiki.openstreetmap.org/wiki/Node)",
+                                            "properties": {
+                                                "id": {
+                                                    "$ref": "#/definitions/node/id"
+                                                },
+                                                "lat": {
+                                                    "$ref": "#/definitions/node/lat"
+                                                },
+                                                "lon": {
+                                                    "$ref": "#/definitions/node/lon"
+                                                },
+                                                "POI_IDs": {
+                                                    "type": "array",
+                                                    "title": "POI_IDs",
+                                                    "description": " Represents the node IDs of the points of interest near a given point (node) on the street(way)",
+                                                    "items": {
+                                                        "$ref": "#/definitions/node/id"
+                                                    }
+                                                }
+                                            }
+                                        }
+                                    }
+                                },
+                                "required": ["street_id"]
+                            }
+                        }
+                    }
+                }
+            },
+            "required": ["street_info"]
```

Comment:
There is no street info key that is ever returned...the returned object just contains streets, POIs, etc.

Reply:
> There is no street info key that is ever returned...the returned object just contains streets, POIs, etc.

I don't seem to understand your comment here. Could you please explain? Thanks

Reply:
RE: old name, it sounds like you're trying to create a new file. Just edit the existing one.

The schema describes the object to be validated. When you put "type: object" on line 4, you are saying that this is meant to validate an object. The start of the properties key on line 30 defines the properties this object must have. When you set the base of this to be "street_info" containing everything else, you are saying the properties field called "ca.mcgill.11y.image.preprocessor.openstreetmaps" is an object with key "street_info" with these other keys (streets, points_of_interest, bounds) within it.